kernel/src/cpu/x86/smap.rs

```diff
@@ -0,0 +1,27 @@
+// SPDX-License-Identifier: MIT
+//
+// Copyright (c) 2024 SUSE LLC
+//
+// Authors: Thomas Leroy <tleroy@suse.de>
+
+use core::arch::asm;
+
+/// Clears RFLAGS.AC to enable SMAP.
+/// This is currently only used when SMAP is supported and enabled.
+/// SMAP protection is effective only if CR4.SMAP is set and if RFLAGS.AC = 0.
+#[inline(always)]
+pub fn clac() {
+    if !cfg!(feature = "nosmap") {
+        unsafe { asm!("clac", options(att_syntax, nomem, nostack)) }
```

Comment:
```suggestion
        unsafe { asm!("clac", options(att_syntax, nomem, nostack, preserves_flags)) }
```

[Inline assembly](https://doc.rust-lang.org/reference/inline-assembly.html) may touch the `AC` flag if the `preserves_flags` is set.

Reply:
I did paid attention to this but I guess I misunderstood the doc:
> The asm! block does not modify the flags register (defined in the rules below). This allows the compiler to avoid recomputing the condition flags after the asm! block.

Isn't it relevant if the code inside the `asm!()` doesn't modify the flags? Which is obviously not the case for `clac/stac`

Reply:
> Maybe it's actually because `stac`/`clac` changes the AC flag, which is not included in the `preserves_flags` rule: CF, PF, AF, ZF, SF, OF

It's this. The compiler doesn't care about the `AC` flag.